app/controllers/markets_controller.rb

```diff
@@ -0,0 +1,107 @@
+class MarketsController < ApplicationController
+
+  before_action :get_market, only: [:edit, :show, :update]
+
+  def get_market
```

Comment:
Since this is used in the `before_action` it might also make sense to put in the `private` section since it shouldn't be used outside the controller class